### ISSUE TITLE
UnboundLocalError: local variable 'data' referenced before assignment

### ISSUE THREAD

**Issue by m1009d** [Juniper/py-junos-eznc#1211]
There is an unhandled case when the `data` variable is not set in the [start_shell.py#L82](https://github.com/Juniper/py-junos-eznc/blob/master/lib/jnpr/junos/utils/start_shell.py#L82) and the script will fail:

```py
Traceback (most recent call last):
  File "/home/mdieska/git/michal_dieska/scripts/dcn/configure_users_from_scratch/configure_users_from_scratch.py", line 458, in configure_users
    result = run_shell_commands(
  File "/home/mdieska/git/michal_dieska/scripts/dcn/configure_users_from_scratch/configure_users_from_scratch.py", line 578, in run_shell_commands
    with StartShell(device, timeout=timeout) as shell:
  File "/home/mdieska/git/venv/lib/python3.10/site-packages/jnpr/junos/utils/start_shell.py", line 202, in __enter__
    self.open()
  File "/home/mdieska/git/venv/lib/python3.10/site-packages/jnpr/junos/utils/start_shell.py", line 125, in open
    got = self.wait_for(r"(%|>|#|\$)")
  File "/home/mdieska/git/venv/lib/python3.10/site-packages/jnpr/junos/utils/start_shell.py", line 81, in wait_for
    if isinstance(data, bytes):
UnboundLocalError: local variable 'data' referenced before assignment
```

**Comment by m1009d** [comment]
Proposed solution:

https://github.com/Juniper/py-junos-eznc/pull/1212#issue-1411565333

**Comment by chidanandpujar** [comment]
Hi ,

Please share the steps to reproduce this issue .

Thanks 
Chidanand

**Comment by ipmonk** [comment]
this looks same as #1203, fixed by #1207
